certbot/main.py

```diff
@@ -87,11 +83,17 @@ def _auth_from_domains(le_client, config, domains, lineage=None):
         # interested in
         action = "renew"
 
+    if action == "renew" and lineage.has_pending_deployment():
```

Comment:
Why should we only do this if `action == "renew"`? Why not do it for all actions? Since the current behavior is to always deploy the certificate, the only way `has_pending_deployment` is `True` is if Certbot crashed. Let's give the user their more or less hidden certificate, help them avoid rate limits, etc.